rating/peersRatingHandler.go

```diff
@@ -110,17 +130,20 @@ func (prh *peersRatingHandler) getOldRating(pid core.PeerID) (int32, bool) {
 }
 
 func (prh *peersRatingHandler) updateRatingIfNeeded(pid core.PeerID, updateFactor int32) {
+	defer prh.updateMetrics()
+
 	oldRating, found := prh.getOldRating(pid)
 	if !found {
 		// new pid, add it with default rating
 		prh.topRatedCache.Put(pid.Bytes(), defaultRating, int32Size)
-		return
+		prh.updateRatingsMap(pid, defaultRating, updateFactor)
```

Comment:
wrongly removed return?

Reply:
+1

Reply:
removed on purpose:
- before: if peer was not in cachers and an IncreaseRating call was made, it would have been added with rating 0
- now: if peer is not in cachers and an IncreaseRating call is made, it will be added with rating 2